src/libstd/sys/windows/mod.rs

```diff
@@ -72,10 +72,13 @@ pub fn decode_error_kind(errno: i32) -> ErrorKind {
     }
 }
 
-fn to_utf16_os(s: &OsStr) -> Vec<u16> {
-    let mut v: Vec<_> = s.encode_wide().collect();
-    v.push(0);
-    v
+pub fn to_u16s(s: &OsStr) -> Option<Vec<u16>> {
```

Comment:
The `fs` module probaby needs its own implementation that takes `Path` instead of `OsString` though, so I figured it might be better this way. Also, it allows for better error messages.